tests/test_ac_harness_shared_memory.py

```
@@ -0,0 +1,205 @@
+
+def test_detector_stuck_false_once_driving():
+    det = DrivingEntryDetector(required_live_reads=1)
+    g = _g(AcGameStatus.LIVE)
+    det.observe(g, PhysicsSnapshot(packet_id=1), now=0.0)
+    assert det.driving is True
+    assert det.stuck_in_menu(g, now=0.0) is False
+
+
+@pytest.mark.parametrize("kwargs", [{"required_live_reads": 0}, {"stagnation_seconds": 0.0}])
+def test_detector_validates_constructor_args(kwargs: dict):
+    with pytest.raises(ValueError):
```

Ruff (0.15.15)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)